bigquery/datasets.js

```diff
@@ -0,0 +1,201 @@
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Found %d dataset(s)!', datasets.length);
+    return callback(null, datasets);
+  });
+}
+// [END list_datasets]
+
+// [START get_dataset_size]
+/**
+ * Calculate the size of the specified dataset.
+ *
+ * @param {string} datasetId The ID of the dataset.
+ * @param {string} projectId The project ID of the dataset.
+ * @param {function} callback The callback function.
+ */
+function getDatasetSize (datasetId, projectId, callback) {
+  // Instantiate a bigquery client
+  var bigquery = BigQuery({
+    projectId: projectId
+  });
+  var dataset = bigquery.dataset(datasetId);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/bigquery/latest/bigquery/dataset
+  dataset.getTables(function (err, tables) {
+    if (err) {
+      return callback(err);
+    }
+
+    return async.map(tables, function (table, cb) {
+      // Fetch more detailed info for each table
+      table.get(function (err, tableInfo) {
+        if (err) {
+          return cb(err);
+        }
+        // Return numBytes converted to Megabytes
+        var numBytes = tableInfo.metadata.numBytes;
+        return cb(null, (parseInt(numBytes, 10) / 1000) / 1000);
+      });
+    }, function (err, sizes) {
+      if (err) {
+        return callback(err);
+      }
+      var sum = sizes.reduce(function (cur, prev) {
+        return cur + prev;
+      }, 0);
+
+      console.log('Size of %s: %d MB', datasetId, sum);
+      return callback(null, sum);
+    });
+  });
+}
+// [END get_dataset_size]
+// [END all]
+
+// The command-line program
+var cli = require('yargs');
+var makeHandler = require('../util').makeHandler;
+
+var program = module.exports = {
+  createDataset: createDataset,
+  deleteDataset: deleteDataset,
+  listDatasets: listDatasets,
+  getDatasetSize: getDatasetSize,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
+
+cli
+  .demand(1)
+  .command('create <name>', 'Create a new dataset.', {}, function (options) {
+    program.createDataset(options.name, options.projectId, makeHandler());
+  })
+  .command('delete <datasetId>', 'Delete the specified dataset.', {}, function (options) {
+    program.deleteDataset(options.datasetId, options.projectId, makeHandler());
+  })
+  .command('list', 'List datasets in the authenticated project.', {}, function (options) {
+    program.listDatasets(options.projectId, makeHandler(true, 'id'));
+  })
+  .command('size <datasetId>', 'Calculate the size of the specified dataset.', {}, function (options) {
+    program.getDatasetSize(options.datasetId, options.projectId, makeHandler());
+  })
+  .option('projectId', {
+    alias: 'p',
+    requiresArg: true,
+    type: 'string',
+    default: process.env.GCLOUD_PROJECT,
+    description: 'Optionally specify the project ID to use.',
+    global: true
+  })
+  .example('node $0 create my-dataset', 'Create a new dataset named "my-dataset".')
```

Comment:
I'm actually removing `projectId` from this particular sample (it doesn't make sense here).